scripts/pipeline_generator/step.py

```diff
@@ -0,0 +1,28 @@
+from pydantic import BaseModel, Field
+from typing import List, Dict, Any, Optional
+
+from .utils import AgentQueue
+
+
+class BuildkiteBlockStep(BaseModel):
+    """This class represents a block step in Buildkite format."""
+    block: str
+    depends_on: Optional[str] = "build"
+    key: str
+
+
+def get_step_key(step_label: str) -> str:
+    step_label = step_label.replace(", ", ",")
+    step_key = ""
+    skip_chars = "()%"
+    for char in step_label.lower():
+        if char in ", ":
+            step_key += "-"
```

Comment:
maybe this should be like "if the last char was `-`, just skip this char"? then you do not need the `replace(", ", ",")`, and can more gracefully handle cases like `,,` or `,  ` (double spaces)?

Reply:
done